File: backend/app/schemas/guardias.py
"""Pydantic schemas for Guardia."""

from pydantic import BaseModel, ConfigDict


class GuardiaCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    asignacion_id: str
    materia_id: str
    carrera_id: str | None = None
    cohorte_id: str | None = None
    dia: str
    horario: str
    comentarios: str | None = None


class GuardiaRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    asignacion_id: str
    materia_id: str
    carrera_id: str | None = None
    cohorte_id: str | None = None
    dia: str
    horario: str
    estado: str
    comentarios: str | None = None
    tenant_id: str

File: frontend/src/features/coordinacion/types/guardias.ts
export interface Guardia {
  id: string
  asignacion_id: string
  materia_id: string
  materia_nombre: string
  carrera_id: string
  cohorte_id: string
  dia: string
  horario: string
  estado: string
  comentarios: string | null
  creada_at: string
  docente: string | null
}

export interface GuardiaPayload {
  materia_id: string
  dia: string
  horario: string
  comentarios?: string | null
}
